### PR TITLE
geo-replication: Error handling and same Volume ID validation

### PR DESCRIPTION
**Pull request by aravindavk**
Updates: #557
Signed-off-by: Aravinda VK <avishwan@redhat.com>

### CODE REVIEW

**Comment by aravindavk** [comment]
@kotreshhr please review

**Comment by aravindavk** [comment]
@kotreshhr fixed now.

File: plugins/georeplication/transactions.go [hunk 1]
```diff
@@ -70,6 +70,11 @@ func gsyncdAction(c transaction.TxnCtx, action actionType) error {
 
 	switch action {
 	case actionStart:
+		// Create Geo-replication Log dir if not exists
+		err = os.MkdirAll(path.Join(config.GetString("logdir"), "glusterfs", "geo-replication"), 755)
```
Comment (prashanthpai) [inline plugins/georeplication/transactions.go]: Shouldn't this be octal (0755) ?

Comment (aravindavk) [reply plugins/georeplication/transactions.go]: Fixed

**Pull request merged**